java/dataset/src/test/java/org/apache/arrow/dataset/file/TestFileSystemDataset.java

```diff
@@ -271,28 +281,34 @@ public void testErrorThrownWhenIterateOnIteratorAfterTaskClose() throws Exceptio
     NativeScanner scanner = dataset.newScan(options);
     List<? extends NativeScanTask> tasks = collect(scanner.scan());
     NativeScanTask task = tasks.get(0);
-    ScanTask.BatchIterator iterator = task.execute();
+    ArrowReader reader = task.execute();
     task.close();
-    assertThrows(NativeInstanceReleasedException.class, iterator::hasNext);
+    assertThrows(NativeInstanceReleasedException.class, reader::loadNextBatch);
+    AutoCloseables.close(factory);
   }
 
+  // After ARROW-7272: It's no longer possible to perform this check as we pre-allocated
+  // container buffers in VectorSchemaRoot
+  @Ignore
```

Comment:
Is there not coverage for file system dataset with this change?

Reply:
This check is not valid for current reader-based API. As Java C data interface doesn't add buffer sizes to the accounting allocator. I think in a way that is reasonable for dataset API because the allocator didn't ever allocate buffers. So I'd remove this test.